bugbot/rules/inactive_revision.py

```diff
@@ -0,0 +1,328 @@
+
+        return bugs
+
+    def load_template(self, template_filename: str) -> Template:
+        env = Environment(loader=FileSystemLoader("templates"))
+        template = env.get_template(template_filename)
+        return template
+
+    def _add_needinfo(self, bugid: str, inactive_revs: list) -> None:
+        has_old_patch = any(
+            revision["created_at"] < self.old_patch_limit for revision in inactive_revs
+        )
+
+        for revision in inactive_revs:
+            last_action_by, _ = self._find_last_action(revision["rev_id"])
+            if last_action_by == "author" and revision["reviewers"]:
+                ni_mail = revision["reviewers"][0]["phab_username"]
+                summary = (
+                    "The last action was by the author, so needinfoing the reviewer."
+                )
+                template = self.ni_reviewer_template
+                nickname = revision["reviewers"][0]["phab_username"]
+            elif last_action_by == "reviewer":
+                ni_mail = revision["author"]["phab_username"]
+                summary = (
+                    "The last action was by the reviewer, so needinfoing the author."
+                )
+                template = self.ni_author_template
+                nickname = revision["author"]["phab_username"]
+            else:
+                continue
+
+            comment = template.render(
+                revisions=[revision],
+                nicknames=nickname,
+                reviewers_status_summary=summary,
+                has_old_patch=has_old_patch,
+                plural=utils.plural,
+                documentation=self.get_documentation(),
+            )
+
+            self.autofix_changes[bugid] = {
+                "comment": {"body": comment},
+                "flags": [
+                    {
+                        "name": "needinfo",
+                        "requestee": ni_mail,
+                        "status": "?",
+                        "new": "true",
+                    }
+                ],
+            }
+
+    def _find_last_action(self, revision_id):
+        details = self._fetch_revisions([revision_id])
+
+        if not details:
+            return None, None
+
+        revision = details[0]
+        author_phid = revision["fields"]["authorPHID"]
+        reviewers = [
+            reviewer["reviewerPHID"]
+            for reviewer in revision["attachments"]["reviewers"]["reviewers"]
+        ]
+
+        transactions = self._fetch_revision_transactions(revision["phid"])
+        last_transaction = transactions[0] if transactions else None
+
+        if last_transaction:
+            last_action_by_phid = last_transaction["authorPHID"]
+            if last_action_by_phid == author_phid:
+                last_action_by = "author"
+            elif last_action_by_phid in reviewers:
+                last_action_by = "reviewer"
+            else:
+                last_action_by = "other"
```

Comment:
There could be cases where the last action is from a bot, e.g. changing some metadata for a set of revisions.

Maybe we should just ignore "other" and only consider who was the last between author and reviewer.